### ISSUE TITLE
data-wb-tables no longer work in v4.0.35

### ISSUE THREAD

**Issue by joejoseph00** [wet-boew/wet-boew#8905]
Reverting back to v4.0.32 fixed the issue:

see the markup:
```
<p>Policies relevant to different branches of our department.</p>

<table class="wb-tables table table-striped table-hover" id="dataset-filter" aria-live="polite" data-wb-tables='{
            "bDeferRender": true,
            "ajaxSource": "/sites/default/files/legacy/resources/prev/doc/ccb_dgcc/policies_v8_EN.json",
            "order": [3, "desc"],
             "columns": [
                { "data": "Branch", "visible": true },
                { "data": "Topic", "visible": true },
                { "data": "Policy", "visible": true },
                { "data": "Date", "visible": true }
              ]}'> 
<thead>
<tr>
<th>Branch/Directorate</th>
<th>Topic</th>
<th>Policy, directive, standard or guideline</th>
<th>Effective date</th>
</tr>
</thead>
</table>

```

**Comment by joejoseph00** [comment]
[wetboew_compatibility.zip](https://github.com/wet-boew/wet-boew/files/4931727/wetboew_compatibility.zip)

**Comment by joejoseph00** [comment]
see the above zip file for the source

**Comment by joejoseph00** [comment]
For those using Drupal I can assist you on downgrading wet-boew.js to v4.0.32

https://www.drupal.org/project/wxt/issues/3159325#comment-13750082

**Comment by thekodester** [comment]
There a new version v4.0.36.

Also not sure why this is not working for your as the demo pages are working without issue.

https://github.com/wet-boew/wet-boew/releases
https://wet-boew.github.io/v4.0-ci/demos/tables/tables-en.html

**Comment by joejoseph00** [comment]
ok I will try 4.0.36 to see if that helps, should be able to test this in the next few days

**Comment by joejoseph00** [comment]
Ok I tried 4.0.36, same exact error

Look at this page:
https://wxt.9pro.ca/en/demo-wet-boewjs-v4035-broken-works-v4032-0

If I switch the library to 4.0.32 , the table loads correctly.
more suscinctly , if I just switch wet-boew.js from 4.0.36 or 4.0.35 to 4.0.32, the table loads correctly

I have not yet tried 4.0.33 or 4.0.34

**Comment by joejoseph00** [comment]
@thekodester 4.0.36 does not work either
https://wxt.9pro.ca/en/demo-wet-boewjs-v4035-broken-works-v4032-0

**Comment by joejoseph00** [comment]
```
error message is as follows:
Line number 11216 of wet-boew.js v4.0.35 OR v4.0.36
```

@thekodester 

https://wxt.9pro.ca/libraries/wet-boew/js/wet-boew.js

```
Uncaught TypeError: $elm.context is undefined
    <anonymous> https://wxt.9pro.ca/libraries/wet-boew/js/wet-boew.js?qdknfv:11216
    jQuery 16
    complete https://wxt.9pro.ca/libraries/wet-boew/js/wet-boew.js?qdknfv:11159
    whenLibReady https://wxt.9pro.ca/libraries/wet-boew/js/wet-boew.js?qdknfv:422
    whenLibReady https://wxt.9pro.ca/libraries/wet-boew/js/wet-boew.js?qdknfv:425

```
https://wxt.9pro.ca/en/demo-wet-boewjs-v4035-broken-works-v4032-0

**Comment by thekodester** [comment]
Thanks for the additional info. I actually run into this my self. The issue is using jQuery v3.4.1, there was recent changed that breaks datatables when using that version of jQuery (other plugins can also break with that version).

There a discussion about this here: https://github.com/wet-boew/wet-boew/issues/8557

**Comment by thekodester** [comment]
TLDR WET has not been tested to work with jQuery v3.4.1 so you are gonna run into issues.

**Comment by joejoseph00** [comment]
@thekodester please note, that I have downgraded to wet-boew.js 4.0.32 and everything works with the same version of jQuery

https://wxt.9pro.ca/en/demo-wet-boewjs-v4035-broken-works-v4032-0

Please visit the same page now, with downgraded wet-boew.
v4.0.32 works

I would still hold this as a bug in wet-boew, something was introduced in newest versions is causing something that used to work fine into something that is now broken.

I understand that you've not tested with all jQuery version that's fine but just FYI

**Comment by thekodester** [comment]
But it not a bug, it a compatibility issue. Like I said there was recent change that breaks it with that version of jQuery, cause the code that was introduced was removed in the newer version of jQuery.

You have two choices downgrade wet-boew version or jQuery. I would downgrade jQuery to what is accepted by WET until work is done to make WET compatible with newer jQuery version.

###### Also I am just a volunteer on this GitHub project.

**Comment by thekodester** [comment]
Thats just my two cents. But downgrading are the only options at this current time until work is done to make newer jQuery work.

**Comment by thekodester** [comment]
This is what is currently accepted by WET `<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js"></script>`

**Comment by joejoseph00** [comment]
"the code that was introduced was removed in the newer version of jQuery."

Ok, I will notify our teams to pin our wet-boew.js to v4.0.32 until further notice.

FYI: I have created a Drupal module that downgrades wet-boew automatically just by enabling it.

If anyone encounters this problem with wxt , clone this project (Drupal 8/9 module) and install it.  the module is called wxt_overrides and includes v4.0.32 wet-boew js

https://github.com/joejoseph00/wxt_overrides

Thanks

**Comment by joejoseph00** [comment]
Definately not going to downgrade to jQuery 2.x , this is out of the question

**Comment by joejoseph00** [comment]
can alternatively downgrade only for certain pages, just add a route check in the module and check for the page.  Probably will do this so that it's only downgraded when necessary.

**Comment by joejoseph00** [comment]
@thekodester  won't fix is a bit harsh unless you plan to switch to esr6,  this should be marked as 'postponed'

In any case, thank you for providing insight into the cause.  

Again, for those that are affected and cannot wait for a fix, just download this module https://github.com/joejoseph00/wxt_overrides/archive/master.zip   extract it as wxt_overrides and install it.

**Comment by duboisp** [comment]
Change request: https://github.com/wet-boew/wet-boew/blob/master/src/plugins/tables/tables.js#L178

```
"href": "#" + $elm.context.id
```

by something like: `$elm.get( 0 ).id`

**Comment by joejoseph00** [comment]
I confirm that $elm.get( 0 ).id works!
Thanks for the fix, I will make a pull request

**Comment by joejoseph00** [comment]
https://github.com/wet-boew/wet-boew/pull/8912  

Merci bcp @duboisp , ça fonctionnent parfaitement selon mes tests!

**Comment by GormFrank** [comment]
Bonjour @joejoseph00, est-ce que cet issue peut être fermée?
